core/lib/snapshots_applier/src/tests/mod.rs

```diff
@@ -142,6 +143,163 @@ async fn snapshots_creator_can_successfully_recover_db(
+    let pool = ConnectionPool::<Core>::test_pool().await;
+    let mut expected_status = mock_recovery_status();
+    expected_status.storage_logs_chunks_processed = vec![true; 10];
+    let storage_logs = random_storage_logs(expected_status.l1_batch_number, 200);
+    let (object_store, client) = prepare_clients(&expected_status, &storage_logs).await;
+    let (stopping_object_store, mut stop_receiver) =
+        HangingObjectStore::new(object_store.clone(), 1);
+
+    let mut config = SnapshotsApplierConfig::for_tests();
+    config.max_concurrency = NonZeroUsize::new(1).unwrap();
+    let task = SnapshotsApplierTask::new(
+        config.clone(),
+        pool.clone(),
+        Box::new(client.clone()),
+        Arc::new(stopping_object_store),
+    );
+    let task_handle = tokio::spawn(task.run());
+
+    // Wait until the first storage logs chunk is requested (the object store hangs up at this point)
+    stop_receiver.wait_for(|&count| count > 1).await.unwrap();
+    assert!(!task_handle.is_finished());
+    task_handle.abort();
+
+    // Check that factory deps have been persisted, but no storage logs.
+    let mut storage = pool.connection().await.unwrap();
+    let all_factory_deps = storage
+        .factory_deps_dal()
+        .dump_all_factory_deps_for_tests()
+        .await;
+    assert!(!all_factory_deps.is_empty());
+    let all_storage_logs = storage
+        .storage_logs_dal()
+        .dump_all_storage_logs_for_tests()
+        .await;
+    assert!(all_storage_logs.is_empty(), "{all_storage_logs:?}");
+
+    // Recover 3 storage log chunks and stop again
+    let (stopping_object_store, mut stop_receiver) =
+        HangingObjectStore::new(object_store.clone(), 3);
+
+    let task = SnapshotsApplierTask::new(
+        config.clone(),
+        pool.clone(),
+        Box::new(client.clone()),
+        Arc::new(stopping_object_store),
+    );
+    let task_handle = tokio::spawn(task.run());
+
+    stop_receiver.wait_for(|&count| count > 3).await.unwrap();
+    assert!(!task_handle.is_finished());
+    task_handle.abort();
+
+    let all_storage_logs = storage
+        .storage_logs_dal()
+        .dump_all_storage_logs_for_tests()
+        .await;
+    assert!(all_storage_logs.len() < storage_logs.len());
+
+    // Recover remaining 7 (10 - 3) storage log chunks.
+    let (stopping_object_store, _) = HangingObjectStore::new(object_store.clone(), 7);
+    let mut task = SnapshotsApplierTask::new(
+        config,
+        pool.clone(),
+        Box::new(client),
+        Arc::new(stopping_object_store),
+    );
+    task.set_snapshot_l1_batch(expected_status.l1_batch_number); // check that this works fine
+    task.run().await.unwrap();
+
+    let all_storage_logs = storage
+        .storage_logs_dal()
+        .dump_all_storage_logs_for_tests()
+        .await;
+    assert_eq!(all_storage_logs.len(), storage_logs.len());
+}
+
+#[tokio::test]
+async fn applier_errors_if_snapshot_l1_batch_differs_from_previous_one() {
```

Comment:
This imo should not be an error as mentioned in previous comments. Lets make this as simple as possible. This is a feature to use mostly for us for some debugging, if partners ever need to use it, they will rather have our guidance and not use two versions